presto-main/src/main/java/com/facebook/presto/split/SplitSourceProvider.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.split;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.metadata.TableLayoutHandle;
+import com.facebook.presto.spi.connector.ConnectorSplitManager.SplitSchedulingStrategy;
+
+public interface SplitSourceProvider
```

Comment:
Do we really need this interface ? Looks like we can just pass `SplitManager` to `CloseableSplitSourceProvider` right ? :D

Reply:
I wanted to encapsulate the `CloseableSplitSourceProvider#close` method from anywhere else, but the place where the `CloseableSplitSourceProvider` is created. So it is easier to reason about the code, as it is guaranteed that the `CloseableSplitSourceProvider` won't be closed anywhere else in the code.